scalardb-test/src/main/java/kelpie/scalardb/transfer/TransferProcessorWithRetry.java

```diff
@@ -0,0 +1,110 @@
+package kelpie.scalardb.transfer;
+
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.DistributedTransactionManager;
+import com.scalar.db.api.Get;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.Result;
+import com.scalar.db.exception.transaction.CommitConflictException;
+import com.scalar.db.exception.transaction.CrudConflictException;
+import com.scalar.db.exception.transaction.RollbackException;
+import com.scalar.kelpie.config.Config;
+import com.scalar.kelpie.modules.TimeBasedProcessor;
+import io.github.resilience4j.core.IntervalFunction;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+import java.util.Optional;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.LongAdder;
+import javax.json.Json;
+
+public class TransferProcessorWithRetry extends TimeBasedProcessor {
+
+  private static final int INITIAL_INTERNAL_MILLS = 10;
+  private static final double MULTIPLIER = 2.0;
+  private static final int MAX_RETRIES = 10;
+
+  private final DistributedTransactionManager manager;
+  private final int numAccounts;
+  private final Retry retry;
+  private final LongAdder retryCount = new LongAdder();
+
+  public TransferProcessorWithRetry(Config config) {
+    super(config);
+    this.manager = TransferCommon.getTransactionManager(config);
+    this.numAccounts = (int) config.getUserLong("test_config", "num_accounts");
+
+    RetryConfig retryConfig =
+        RetryConfig.custom()
+            .maxAttempts(MAX_RETRIES)
+            .intervalFunction(
+                IntervalFunction.ofExponentialBackoff(INITIAL_INTERNAL_MILLS, MULTIPLIER))
+            .retryExceptions(CrudConflictException.class, CommitConflictException.class)
+            .build();
+    retry = Retry.of("TransferProcessorWithRetry", retryConfig);
+    retry
+        .getEventPublisher()
+        .onRetry(
+            e -> {
+              logWarn(e.toString(), e.getLastThrowable());
+              retryCount.increment();
+            });
```

Comment:
Use `EventPublisher.onRetry()` to log the exception and increment the retry count.